ci/base.yml

```diff
@@ -3,47 +3,68 @@ include:
 
 stages:
   - baseimage
-  - build
+  - image
   - test
   - benchmark
 
-.py310: &py310
+variables:
   PYVERSION_PREFIX: py310
   PYVERSION: 3.10.9
 
 # Base image build step with SHA256 checksum for caching
-build_baseimage:
-  extends: .container-builder-cscs-zen2
+.build_baseimage:
   stage: baseimage
   before_script:
-    - DOCKER_TAG=`sha256sum $DOCKERFILE | head -c 16`
-    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/public/base/icon4py:$DOCKER_TAG-$PYVERSION
+    # include build arguments in hash since we use a parameterized Docker file
+    - DOCKER_TAG=`echo "$(cat $DOCKERFILE) $DOCKER_BUILD_ARGS" | sha256sum | head -c 16`
+    - export PERSIST_IMAGE_NAME=$CSCS_REGISTRY_PATH/public/$ARCH/base/icon4py:$DOCKER_TAG-$PYVERSION
     - echo "BASE_IMAGE_${PYVERSION_PREFIX}=$PERSIST_IMAGE_NAME" >> build.env
   artifacts:
     reports:
       dotenv: build.env
   variables:
     DOCKERFILE: ci/docker/base.Dockerfile
+    # change to 'always' if you want to rebuild, even if target tag exists already (if-not-exists is the default, i.e. we could also skip the variable)
     CSCS_REBUILD_POLICY: if-not-exists
-    DOCKER_BUILD_ARGS: '["PYVERSION=$PYVERSION", "CI_PROJECT_DIR=$CI_PROJECT_DIR"]'
-    <<: *py310
+    DOCKER_BUILD_ARGS: '["ARCH=$ARCH", "BASE_IMAGE=$BASE_IMAGE", "HPC_SDK_VERSION=$HPC_SDK_VERSION", "HPC_SDK_NAME=$HPC_SDK_NAME", "CUPY_PACKAGE=$CUPY_PACKAGE", "PYVERSION=$PYVERSION", "CI_PROJECT_DIR=$CI_PROJECT_DIR"]'
+build_baseimage_x86_64:
```

Comment:
Once daint is decomissioned will we still be able to run tests for a machine with x86_64 architecture?

Reply:
I know there is a vCluster with AMD CPUs (x86_64) but I don't know if there are vCluster with x86_64 + GPU nodes.